source/MaterialXTest/ShaderValid.cpp

```diff
@@ -303,22 +309,43 @@ static mx::GlslValidatorPtr createValidator(bool& orthographicView, const std::s
 //
 // Outputs error log if validation fails
 //
-static void runValidation(const std::string& outputPath, const std::string& shaderName, mx::ElementPtr element,
-                          mx::GlslValidatorPtr validator, mx::ShaderGeneratorPtr shaderGenerator,
-                          bool orthographicView, mx::DocumentPtr doc, std::ostream& log, bool outputMtlxDoc=false)
+static void runValidation(const std::string& shaderName, mx::ElementPtr element, mx::GlslValidatorPtr validator,
+                          mx::ShaderGeneratorPtr shaderGenerator, bool orthographicView, mx::DocumentPtr doc,
+                          std::ostream& log, bool outputMtlxDoc=true, const std::string& outputFolder="",
```

Comment:
The GlslValidatorPtr could be a ref instead of a pointer, since it is assumed below that it's non-null. Same thing with the ShaderGeneratorPtr.

Also, since the GlslValidator is already limiting the function to do only GLSL validation the ShaderGenerator could be a GlslShaderGenerator, and the dynamic_cast below could be removed.

Reply:
Will do.

Reply:
GlslShaderGenerator generate still generates a ShaderPtr and not a HwShaderPtr, so I'd think we'd still need a cast of some sort, no?

Reply:
If it's a GlslShaderGenerator then you already know it will generate a GlslShader which is a HwShader (even if it's stored in a ShaderPtr), so you don't need to cast it to check that. The only method called on it below is getSourceCode() which is available from the ShaderPtr.